schema_based_on_diagram.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TABLE patients (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  name VARCHAR(100) NOT NULL,
+  date_of_birth DATE NOT NULL,
+);
+
+CREATE TABLE medical_histories (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  admitted_at TIMESTAMP NOT NULL,
+  patient_id INT NOT NULL,
+  status VARCHAR(50),
+  CONSTRAINT fk_patient_id
+    FOREIGN KEY (patient_id)
+    REFERENCES patients(id)
+);
+
+CREATE TABLE invoices (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  total_amount DECIMAL NOT NULL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP NOT NULL,
+  medical_history_id INT NOT NULL,
+  CONSTRAINT fk_medical_history_id
+    FOREIGN KEY (medical_history_id)
+    REFERENCES medical_histories(id)
+);
+
+CREATE TABLE treatments (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  type VARCHAR(50),
+  name VARCHAR(100)
+);
+
+CREATE TABLE invoice_items (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  unit_price DECIMAL NOT NULL,
+  quantity INT NOT NULL,
+  total_price DECIMAL NOT NULL,
+  invoice_id INT NOT NULL,
+  treatment_id INT NOT NULL,
+  CONSTRAINT fk_invoice_id
+    FOREIGN KEY (invoice_id)
+    REFERENCES invoices(id),
+  CONSTRAINT fk_treatment_id
+    FOREIGN KEY (treatment_id)
+    REFERENCES treatments(id)
+);
+
```

Comment:
- [ ] I could see here you didn't add any indexes so I highly recommend adding indexes to all FK as mentioned in the requirements `Remember to add the FK indexes.` because the indexes are very important Indexes are used quickly to locate data without having to search every row in a database table every time said table is accessed. 💯🤩 you can check this [lesson](https://www.w3schools.com/sql/sql_create_index.asp) to learn more about how to create indexes .✨

![image](https://github.com/Movgto/database-from-diagram/assets/58553711/2f1aeaad-7b80-4ad8-b5a6-b8ad0f002454)